### PR TITLE
73 sections arent being garbage collected

### PR DESCRIPTION
**Pull request by edovanveen**

### CODE REVIEW

**Review by AljenU — APPROVED** [review]

**Comment by edovanveen** [comment]
@ramcdougal can I merge this one?

File: +neuron/Section.m [hunk 1]
```diff
@@ -52,19 +65,24 @@
                 error("Initialize a Neuron session before making a Section.");
             end
         end
+        
+        function delete_nrn_sec(self)
+        % Destroy the Neuron Section.
+        %   delete_nrn_sec()
+            % clib.neuron.section_unref(self.sec);
+            self.push();
+            clib.neuron.hoc_oc("delete_section()");  % Annoyingly, this outputs '1' to the console.
```
Comment (ramcdougal) [inline +neuron/Section.m:74]: To suppress printing, add curly braces:

```matlab
 clib.neuron.hoc_oc("{delete_section()}");
```

**Review by ramcdougal — CHANGES_REQUESTED** [review]
There's one specific change that's needed to suppress printing. (add curly braces, see comment).

Because of the all_sections test and because sections are now handles, I think this should work, but can you confirm that something like:
```matlab
axon = n.Section("axon");
axon2 = axon;
axon = 42;
axon2.L = 47;
```
still works? That is, as long as there is a way to get to the section, the destructor should not be called.

**Comment by edovanveen** [comment]
@ramcdougal 

This seems to work okay:

```matlab
>> n = neuron.Neuron();
>> axon = n.Section("axon");
>> axon2 = axon;
>> axon.length = 42;
>> axon.length

ans =

    42

>> axon2.length = 72;
>> axon.length

ans =

    72

>> clear axon;
>> axon2.length

ans =

    72

>> clear axon;
>> n.topology();

|-|       axon(0-1)

>> clear axon2;
>> n.topology();


```

**Pull request merged**